### ISSUE TITLE
Add user-friendly names for datasets?

### ISSUE THREAD

**Issue by sshah-wework** [MarquezProject/marquez#325]
URNs are good for canonical and unique naming, but less friendly for visualization since not everyone thinks about their datasets as URNs (though machines will happily consume them). The URN can be globally unique, but dataset name need only be unique per data source.

For example, given a redshift table `foo.bar`, the URN might look like `:redshift:foo.bar:`. However, for visualization, we may want to present the dataset as `<datasource, dataset_name>`. In other words:

| URN |
| -- |
| :redshift:foo.bar: |
| :redshift:foo.baz: |
| :hive:baz.baz: |

vs. 

| Data Source | Dataset |
| -- | -- |
| Redshift | foo.bar |
| Redshift | foo.baz |
| Hive | baz.baz |

An added benefit of breaking data source as a column is a more natural workflow for filtering datasets by data source.

**Comment by wslulciuc** [comment]
Would this replace, or compliment the `urn` column in the `datasets` table?

As an alternative, given some URN, DatasetAPI could provide a more friendlier response i.e. `datasource`, `dataset_name`, `latest_version`, etc?

**Comment by wslulciuc** [comment]
Thinking about this more, I like the idea of adding the data source as a column in`datasets`:

```sql
CREATE TABLE datasets (
  uuid                 UUID PRIMARY KEY,
  created_at           TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
  updated_at           TIMESTAMPTZ,
  namespace_uuid       UUID REFERENCES namespaces (uuid),
  name                 VARCHAR(64) NOT NULL,
  data_source          data_sources NOT NULL,
  urn                  VARCHAR(64) NOT NULL,
  description          TEXT,
  current_version_uuid UUID,
  UNIQUE (namespace_uuid, urn)
);
```
where, `data_sources` can take on the values `Postgres`, `Redshift`, `MySql`, etc. We would still need to keep track of `connection_url` => `data_source`. The idea was to use the `data_sources` for that. Thoughts?

**Comment by sshah-wework** [comment]
> Would this replace, or compliment the `urn` column in the `datasets` table?

It wouldn't be a replacement to URN, just an additional field.

> 
> As an alternative, given some URN, the DatasetAPI could provide a more friendlier response i.e. `datasource`, `dataset_name`, `latest_version`, etc? Right now, we only respond back with the `urn`, but that's not all that helpful

Yes exactly, but I don't think we should do that by parsing URN... if we had `dataset_name` as a separate field we could just return it separately.

**Comment by sshah-wework** [comment]
> Thinking about this more, I like the idea of adding the data source as a column in`datasets`:
> 
> ```sql
> CREATE TABLE datasets (
>   uuid                 UUID PRIMARY KEY,
>   created_at           TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
>   updated_at           TIMESTAMPTZ,
>   namespace_uuid       UUID REFERENCES namespaces (uuid),
>   name                 VARCHAR(64) NOT NULL,
>   data_source          data_sources NOT NULL,
>   urn                  VARCHAR(64) NOT NULL,
>   description          TEXT,
>   current_version_uuid UUID,
>   UNIQUE (namespace_uuid, urn)
> );
> ```

We have `datasource_uuid` in `datasets`

> where, `data_sources` can take on the values `Postgres`, `Redshift`, `MySql`, etc. We would still need to keep track of `data_source` **=>** `connection_url`. The idea was to use the `data_sources` table for that. Thoughts?

That field should be in `datasources` rather than datasets since every dataset has a datasource. And there will be one datasource for many datasets.

**Comment by sshah-wework** [comment]
I guess I'm not sure what the benefit is of moving "data source type" from datasources to datasets?

**Comment by wslulciuc** [comment]
I see. It would only make sense if we dropped the `data_sources` table. Which is what I thought you were proposing.

**Comment by sshah-wework** [comment]
Ah no, this is a proposal for a new user-friendly `name` field in the `datasets` table

**Comment by sshah-wework** [comment]
Sorry for the confusion there

**Comment by wslulciuc** [comment]
Fixed #627
